src/main/java/com/nike/cerberus/client/HttpClientFactory.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2017 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import com.nike.cerberus.ConfigConstants;
+import com.nike.cerberus.domain.environment.CertificateInformation;
+import com.nike.cerberus.module.CerberusModule;
+import com.nike.cerberus.store.ConfigStore;
+import com.nike.vault.client.StaticVaultUrlResolver;
+import com.nike.vault.client.auth.DefaultVaultCredentialsProviderChain;
+import okhttp3.OkHttpClient;
+import okio.Buffer;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Named;
+import javax.net.ssl.*;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class HttpClientFactory {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    public static final int DEFAULT_TIMEOUT = 15;
+    public static final TimeUnit DEFAULT_TIMEOUT_UNIT = TimeUnit.SECONDS;
+
+    private final ObjectMapper objectMapper;
+    private final ConfigStore configStore;
+
+    @Inject
+    public HttpClientFactory(@Named(CerberusModule.CONFIG_OBJECT_MAPPER) ObjectMapper objectMapper,
+                             ConfigStore configStore) {
+
+        this.objectMapper = objectMapper;
+        this.configStore = configStore;
+    }
+
+    /**
+     * Admin client for doing admin cms tasks
+     */
+    public CerberusAdminClient createCerberusAdminClient(String url) {
```

Comment:
nitpick: Shouldn't this method move to a CerberusAdminClientFactory class?

Reply:
It seems weird as-is.  I'd move the method out or rename the class and make the other factory methods private, if that is possible.